docs/ref/settings.txt

```diff
@@ -687,13 +689,16 @@ When :setting:`USE_TZ` is ``False``, it is an error to set this option.
     third-party systems connect to the same database and expect to find
     datetimes in local time, then you must set this option.
 
-* If the database backend supports time zones (e.g. PostgreSQL), the
-  ``TIME_ZONE`` option is very rarely needed. It can be changed at any time;
-  the database takes care of converting datetimes to the desired time zone.
+* If the database backend supports time zones (e.g., PostgreSQL), then the
+  database connection's time zone is set to this value.
 
-  Setting the time zone of the database connection may be useful for running
-  raw SQL queries involving date/time functions provided by the database, such
-  as ``date_trunc``, because their results depend on the time zone.
+  Although setting the ``TIME_ZONE`` option is very rarely needed it may be
+  necessary to set this to the same value as the general :setting:`TIME_ZONE`
+  setting when raw queries that involve date/time functions such as
+  PostgreSQL's ``date_trunc()`` or ``generate_series()`` when generating
+  time-based series that transition daylight savings. It can be changed at any
+  time; the database takes care of converting datetimes to the desired time
+  zone.
```

Comment:
This fragment seems like it has a grammatical issue: "_when raw queries that involve date/time functions ... when generating time-based series that transition daylight savings_"

Did you mean "_when raw queries that involve date/time functions ... generate time-based series that transition daylight savings_"?

What about:
```suggestion
  Although setting the ``TIME_ZONE`` option is very rarely needed, there are
  situations where it becomes necessary. Specifically, it should match the
  general :setting:`TIME_ZONE` setting when dealing with raw queries involving
  date/time functions like PostgreSQL's ``date_trunc()`` or
  ``generate_series()``, especially when generating time-based series that
  transition daylight savings.

  This value can be changed it at any time, the database will handle
  the conversion of datetimes to the configured time zone.
```

Reply:
lol I just re-read my sentence and it definitely is a challenge because it's too long and has "… when … when …" which is bad.

I like how you split it up 👍 

Re "Specifically, it should match … " I tried to avoid "should" style because it's not a hard rule.  I was going for "may be necessary" or "recommended". Otherwise your suggestion is perfect!

Reply:
Ok! So, how about:

```suggestion
  Although setting the ``TIME_ZONE`` option is very rarely needed, there are
  situations where it becomes necessary. Specifically, it's recommended to
  match the general :setting:`TIME_ZONE` setting when dealing with raw queries
  involving date/time functions like PostgreSQL's ``date_trunc()`` or
  ``generate_series()``, especially when generating time-based series that
  transition daylight savings.

  This value can be changed it at any time, the database will handle
  the conversion of datetimes to the configured time zone.
```